Author: Tony Espy <espy@canonical.com>
Subject: Set IP6_CONFIG_METHOD_IGNORE for ofono connections
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/network-manager/+bug/1444162

This patch modifies the ofono settings plugin to add IP6_CONFIG
settings that contain an IP6_CONFIG_METHOD set to ignore indicate
to NM that IPv6 is not required.  This fixes an issue with QNetwork
which incorrectly assumed that the ofono mobile data connection
support IPv6.

---
 src/settings/plugins/ofono/parser.c |   15 +++++++++++++++
 1 file changed, 15 insertions(+)

Index: b/src/settings/plugins/ofono/parser.c
===================================================================
--- a/src/settings/plugins/ofono/parser.c
+++ b/src/settings/plugins/ofono/parser.c
@@ -49,6 +49,7 @@ ofono_update_connection_from_context (NM
 {
 	NMSettingConnection *s_con;
 	NMSettingGsm *s_gsm;
+	NMSettingIP6Config *s_ip6_config;
 	gboolean success = FALSE;
 	char *idstr = NULL;
 	char *uuid_base = NULL;
@@ -89,6 +90,20 @@ ofono_update_connection_from_context (NM
 	 */
 	g_object_set (s_gsm, NM_SETTING_GSM_NUMBER, "*99#", NULL);
 
+	/* IP6 setting */
+	s_ip6_config = NM_SETTING_IP6_CONFIG (nm_setting_ip6_config_new());
+	g_assert (s_ip6_config);
+	nm_connection_add_setting (connection, NM_SETTING (s_ip6_config));
+
+	/*
+	 * Set IP6_CONFIG_METHOD to ignore to prevent NM from configuring IPv6
+	 * for ofono connections.
+	 */
+	g_object_set (s_ip6_config,
+	              "method",
+	              NM_SETTING_IP6_CONFIG_METHOD_IGNORE,
+	              NULL);
+
 	nm_log_info (LOGD_SETTINGS, "SCPlugin-Ofono: "
 	             "update_connection_setting_from_context: name:%s, path:%s, id:%s, uuid: %s",
 	             (char *) g_hash_table_lookup (context, "Name"),
